packages/lesswrong/components/form-components/FMCrosspostControl.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { useState, useEffect } from "react";
+import { Components, registerComponent } from "../../lib/vulcan-lib";
+import {
+  fmCrosspostSiteNameSetting,
+  fmCrosspostBaseUrlSetting,
+} from "../../lib/instanceSettings";
+import { useSingle } from "../../lib/crud/withSingle";
+import { useCrosspostApolloClient } from "../hooks/useCrosspostApolloClient";
+import FormControlLabel from "@material-ui/core/FormControlLabel";
+import Checkbox from "@material-ui/core/Checkbox";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+import LoginIcon from "@material-ui/icons/LockOpen"
+import UnlinkIcon from "@material-ui/icons/RemoveCircle";
+import classNames from "classnames";
+import { gql, useMutation } from "@apollo/client";
+import { useCurrentUser } from "../common/withUser";
+import { useOnTabView } from "../hooks/useOnTabView";
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    display: "flex",
+    flexDirection: "column",
+    margin: 8,
+  },
+  link: {
+    color: theme.palette.primary.main,
+  },
+  crosspostMessage: {
+    marginBottom: 12,
+  },
+  button: {
+    background: theme.palette.buttons.imageUpload.background,
+    "&:hover": {
+      background: theme.palette.buttons.imageUpload.hoverBackground,
+    },
+    color: theme.palette.text.invertedBackgroundText,
+  },
+  buttonIcon: {
+    fontSize: 18,
+    marginRight: theme.spacing.unit,
+  },
+  error: {
+    color: theme.palette.error.main,
+    margin: "8px 0",
+  },
+});
+
+const FMCrosspostAccount = ({fmCrosspostUserId, classes}: {
+  fmCrosspostUserId: string,
+  classes: ClassesType,
+}) => {
+  const apolloClient = useCrosspostApolloClient();
+  const {document, refetch, loading: loadingDocument} = useSingle({
+    documentId: fmCrosspostUserId,
+    collectionName: "Users",
+    fragmentName: "UsersCrosspostInfo",
+    apolloClient,
+  });
+
+  const link = `${fmCrosspostBaseUrlSetting.get()}users/${document?.slug}`;
+
+  const {Loading} = Components;
+  return document
+    ? (
+      <div className={classes.crosspostMessage}>
+        This post will be crossposted to {fmCrosspostSiteNameSetting.get()} by
+        your account <a className={classes.link} href={link} target="_blank" rel="noreferrer">
+          {document.username}
+        </a>
+      </div>
+    )
+    : (
+      <Loading />
+    );
+}
+
+const FMCrosspostView = ({fmCrosspostUserId, loading, onClickLogin, onClickUnlink, classes}: {
```

Comment:
Just jumping into this file, I find it hard to understand what FMCrosspostView, FMCrosspostAccount, FMCrosspostControl all do and how they relate. Different names or some documentation up top would be appreciated (though don't treat as super blocking)